provider-ci/internal/pkg/templates/bridged-provider/Makefile

```
@@ -258,7 +258,7 @@ bin/$(PROVIDER): .make/schema
 
 test: export PATH := $(WORKING_DIR)/bin:$(PATH)
 test:
-	cd examples && go test -v -tags=all -parallel $(TESTPARALLELISM) -timeout 2h
+	cd #{{ .Config.E2ETestDirectory }}# && go test -v -tags=all -parallel $(TESTPARALLELISM) -timeout 2h
```

Alternative design - proxy all test runs through your sharded test runner which could then apply the sharding in a single step, if needed:

```suggestion
	cd #{{ .Config.E2ETestDirectory }}# && go run github.com/pulumi/go-test-shard@v1 --total "$(SHARD_TOTAL)" --index "$(SHARD_INDEX)" -- -v -tags=all -parallel $(TESTPARALLELISM) -timeout 2h
```

The exact same change could be applied to the `provider_test` target too and could be moved from the prerequisites to the test matrix too.

---

> The exact same change could be applied to the provider_test target too and could be moved from the prerequisites to the test matrix too.

I've been thinking of `provider_test` as essentially unit tests which make sense to keep in `prerequisites` as a lightweight sanity check before the heavy tests start.

See my comment [here](https://github.com/pulumi/ci-mgmt/pull/1213#issuecomment-2540167749) about integration tests in `provider`. Basically our poor module structure is pushing us towards some awkward conventions, and there are a ton of benefits to fixing the underlying issue instead of catering to those conventions.

_By default_ we can expect "slow" tests to live in one directory, but this doesn't have to be a requirement as long as the provider can eventually customize targets. With a root-level `go.mod` it becomes trivial to run integration tests across any number of directories -- in other words it's possible to specify `SHARD_PATHS="./examples ./provider"` when they belong to the same module.